### DISCUSSION TITLE
Mouse, keyboard doesn't work on background

### DISCUSSION THREAD

**Discussion by aca** [go-rod/rod#950] [Q&A]
Hi, I'm trying to connect to current browser with webdriver.
This code only works if browser is available on the screen. 
If my editor/terminal is on full screen, mouse/keyboard control doesn't work (with no error).
I'm using linux, sway(wayland).

How can i use mouse/keyboard if it's not available on the screen?

```
package main

import (
	"github.com/go-rod/rod"
	"github.com/go-rod/rod/lib/input"
	"github.com/go-rod/rod/lib/launcher"
)


var browser *rod.Browser

func main(){
	u := launcher.MustResolveURL("")
	browser = rod.New().ControlURL(u).MustConnect()

	page := browser.MustPage("https://news.ycombinator.com/item?id=37733036")
	err := page.WaitLoad()
	if err != nil {
		panic(err)
	}
	for i := 0; i < 1000; i++ {
		page.KeyActions().Type(input.PageDown).MustDo()
	}
}
```

**Comment by ysmood** [comment]
`page.Mouse.MustScroll`

**Reply by aca** [reply]
```
func main(){
	u := launcher.MustResolveURL("")
	browser := rod.New().ControlURL(u).MustConnect()

	page := browser.MustPage("https://news.ycombinator.com/item?id=37733036")
	err := page.WaitLoad()
	if err != nil {
		panic(err)
	}
	log.Println("scroll start") 
	page.Mouse.MustScroll(0, 100000) // block here
	log.Println("scroll end")
}
```

Same.
This process is blocked when browser is not visible on the screen. 
Mustscroll returns when I close full mode of the terminal running this code and browser is visible.
Code runs fine even if it's not the active window, but doesn't work when it's not actually visible on the monitor.
I'm noob here. Does Mouse, Keyboard actually clicks through the window manager or something like that?

**Reply by ysmood** [reply]
check the ci of rod, it uses linux to do click etc, works fine to us. CI also tests rod in docker.
